### ISSUE TITLE
Bundling multiple independent cases in a single node on the NERSC clusters?

### ISSUE THREAD

**Issue by mpaiao** [ESMCI/cime#4295]
I am not sure if this issue belongs to CIME, E3SM, or FATES as the three are involved. I am trying to run a parameter sensitivity analysis with E3SM-FATES on cori (multiple single site runs). Because each simulation is very light, I would like to have multiple runs going in the same node, but I am running into some problems. 

First, I tried to use `./create_newcase --ninst=XX --multi-driver`. I am able to get the simulations running, but some parameter combinations cause the model to fail. If one of these combinations fail, then the model stops for all the ensemble members. 

As an alternative, I am trying to create multiple cases that I can run independently, so if one ensemble member fails, the others can continue. To do this on core, I generated a series of cases (using a combination of `create_newcase` and `create_clone`), and I wrote a batch shell script that then calls 32 (cori-haswell) or 64 (cori-knl) of these cases for each node, using `case.submit --no-batch`. However,  `case.submit --no-batch` still calls `srun` on cori, and this is causing either only one ensemble member per node to run at any given time, or for all the runs to never start because `srun` cannot find enough resources. I tried to bundle the runs using both  `srun <script that calls case.submit --no-batch>` or using the taskfarmer module available on cori.  

Is there a way to bypass the `srun` call in `case.submit`, or is there a way to make the multi-instance set up to not crash if only one of the instances fail?

**Comment by jedwards4b** [comment]
The multi instance will always crash if a single member fails.  

You can use the create_clone --ensemble to create the cases.  Then I think you need to get an interactive login using https://docs.nersc.gov/jobs/interactive/#cori-haswell
and once you are logged into the node then you can use case.submit with the --no-batch command.

**Comment by jedwards4b** [comment]
I think you will need some custom scripts for what you want to do.   Instead of trying to run ./case.submit --no-batch I would modify template.case.run (it's in the machines directory) to call case.case_run for all of your runs instead of just a single one.  So you would have a list of caseroots and loop over the list starting each in a new thread something like:

**Comment by djk2120** [comment]
With CLM at NCAR, we use casper for single point simulations as an easier way to get just 1 processor for each simulation. So we haven't explicitly been forced to solve this sort of thing.

**Comment by ekluzek** [comment]
@djk2120 with the PPE work, you do send off a long list of simulations all at once. At least that's my understanding. You have one case that you clone the others from, and then have modifications for each of the cloned cases. How do you set it up to run through your list of simulations at one go? I know you have some scripts to manage all of this together.

**Comment by djk2120** [comment]
Just a shell script to submit them independently. Loop through a bunch of paramfiles, cloning the basecase, pointing to the appropriate paramfile in user_nl_clm. I was at one point using multi-instance, but likewise abandoned because it was too cumbersome when a rogue ensemble member failed.

If Marcos can find a way to request just 1 cpu at a time on the DOE system, as we can with casper, that would probably be the easiest solution.

**Comment by mpaiao** [comment]
Thanks @jedwards4b @ekluzek and @djk2120 for the input! After a few additional attempts, I managed to submit multiple independent simulations sharing a single node on cori. In the end the solution was simple, I just had to switch the MPI libraries from default to mpi-serial before I built each case. 

```
./create_newcase <new case settings>
cd <path to new case>

./xmlchange MPILIB="mpi-serial"
./case.setup
./case.build
```

And then I bundled the multiple independent cases using [NERSC's TaskFarmer](https://docs.nersc.gov/jobs/workflow/taskfarmer/) workflow, by requesting 1 one for TaskFarmer and 1 node for every 64 ensemble members on cori-knl (or 1 node for every 32 ensemble members if using cori-haswell).
